### ISSUE TITLE
Tmod4 backend crashes when an unexpected installation is found

### ISSUE THREAD

**Issue by vkarak** [eth-cscs/reframe#781]
How to reproduce on Arolla:

```
ln -s /cm/local/apps/environment-modules/4.0.0/libexec/modulecmd.tcl modulecmd
PATH=.:$PATH ./bin/reframe -C config/cscs.py --system=generic -l
 PATH=.:$PATH ./bin/reframe -C config/cscs.py --system=generic -l
Command line: ./bin/reframe -C config/cscs.py --system=generic -l
Reframe version: 2.18-dev2
Launched by user: karakasv
Launched on host: arolla-ln003
Reframe paths
=============
    Check prefix      : /users/karakasv/Devel/reframe
(R) Check search path : 'checks/'
    Stage dir prefix     : /users/karakasv/Devel/reframe/stage/
    Output dir prefix    : /users/karakasv/Devel/reframe/output/
    Perf. logging prefix : /users/karakasv/Devel/reframe/perflogs
./bin/reframe: unexpected error: '_mlstatus'
Traceback (most recent call last):
  File "/users/karakasv/Devel/reframe/reframe/frontend/cli.py", line 464, in main
    rt.modules_system.unload_module(settings.reframe_module)
  File "/users/karakasv/Devel/reframe/reframe/core/modules.py", line 210, in unload_module
    self._unload_module(m)
  File "/users/karakasv/Devel/reframe/reframe/core/modules.py", line 213, in _unload_module
    self._backend.unload_module(Module(name))
  File "/users/karakasv/Devel/reframe/reframe/core/modules.py", line 463, in unload_module
    msg="could not unload module '%s' correctly" % module)
  File "/users/karakasv/Devel/reframe/reframe/core/modules.py", line 512, in _exec_module_command
    if not namespace['_mlstatus']:
KeyError: '_mlstatus'
```

It shouldn't crashed, but give an error instead.

That's for an older Tmod4 installation:
```
Modules Release 4.0.0 (2017-10-16)
```

**Comment by vkarak** [comment]
ReFrame seems to correctly recognise the Tmod4 installation and the error happens when it tries to unload the `reframe` module at the beginning. It seems that `_mlstatus` is not present in early Tmod4 versions.

**Comment by vkarak** [comment]
It appears that what we are looking for is only available from Tmod 4.1.0 onwards. Quoting from the 4.1.0 release notes:

> Adapt module function definition for Tcl, Perl, Python, Ruby, CMake and R scripting languages to always return a value, result of the modulecmd.tcl run. When modulecmd.tcl run does not produce a specific status, a 'true' boolean value is returned. On CMake, resulting value is returned though a module_result global variable.

Also the `modulecmd` wrapper to `modulecmd.tcl` was again introduced in Tmod 4.1.0:

> Introduce modulecmd wrapper script, installed in binary directory, which executes the active module command.

**Comment by vkarak** [comment]
I think we can quickly fix this issue by do not allowing ReFrame to run on Tmod < 4.1. We should also update the documentation.
